packages/vats/src/vat-bank.js

```diff
@@ -171,6 +171,138 @@ export function buildRootObject(_vatPowers) {
+
+        /** @type {Bank} */
+        const bank = Far('bank', {
+          getAssetSubscription() {
+            return harden(assetSubscription);
+          },
+          async getPurse(brand) {
+            if (brandToVPurse.has(brand)) {
+              return brandToVPurse.get(brand);
+            }
+
+            const assetRecord = brandToAssetRecord.get(brand);
+            if (!bankCall) {
+              // Just emulate with a real purse.
+              const purse = await E(assetRecord.issuer).makeEmptyPurse();
+              brandToVPurse.init(brand, purse);
+              return purse;
+            }
+
+            const addressToUpdater = denomToAddressUpdater.get(
+              assetRecord.denom,
+            );
+
+            /** @type {NotifierRecord<Amount>} */
+            const { updater, notifier } = makeNotifierKit();
+            /** @type {bigint} */
+            let lastBalanceUpdate = -1n;
+            /** @type {BalanceUpdater} */
+            const balanceUpdater = (value, nonce = undefined) => {
+              if (nonce !== undefined) {
+                const thisBalanceUpdate = BigInt(nonce);
+                if (thisBalanceUpdate <= lastBalanceUpdate) {
+                  return;
+                }
+                lastBalanceUpdate = thisBalanceUpdate;
+              }
+              // Convert the string value to a bigint.
+              const amt = amountMath.make(brand, BigInt(value));
+              updater.updateState(amt);
+            };
+
+            // Get the initial balance.
+            addressToUpdater.init(address, balanceUpdater);
+            const balanceString = await bankCall({
+              type: 'VPURSE_GET_BALANCE',
+              address,
+              denom: assetRecord.denom,
+            });
+            balanceUpdater(balanceString);
+
+            // Create and return the virtual purse.
+            const vpc = makePurseController(
+              bankCall,
+              assetRecord.denom,
+              brand,
+              address,
+              notifier,
+              updateBalances,
+            );
+            const vpurse = makeVirtualPurse(vpc, assetRecord);
+            brandToVPurse.init(brand, vpurse);
+            return vpurse;
+          },
+        });
+        addressToBank.init(address, bank);
+        accountsUpdater.updateState([...addressToBank.keys()]);
+        return bank;
+      };
+
+      const bankDepositFacet = Far('bankDepositFacet', {
+        getAccountsNotifier() {
+          return accountsNotifier;
+        },
+        /**
+         * Send many independent deposits.  If any of them fail, then you should
+         * reclaim the corresponding payments since they didn't get deposited.
+         *
+         * @param {Array<string>} accounts
+         * @param {Array<Payment>} payments
+         * @returns {Promise<PromiseSettledResult<Amount>[]>}
+         */
+        depositMultiple(accounts, payments) {
```

Comment:
We should require these are all for the same brand, and provide the alleged brand as an argument?